src/Crypt32.Shared/Crypt32.cs

```diff
@@ -34,6 +34,47 @@ public static partial class Crypt32
+        /// </param>
+        /// <param name="phCryptProvOrNCryptKey">
+        /// The address of an HCRYPTPROV_OR_NCRYPT_KEY_HANDLE variable that receives the handle of either the CryptoAPI provider or the CNG key. If the <paramref name="pdwKeySpec"/> variable receives the CERT_NCRYPT_KEY_SPEC flag, this is a CNG key handle of type NCRYPT_KEY_HANDLE; otherwise, this is a CryptoAPI provider handle of type HCRYPTPROV.
+        /// For more information about when and how to release this handle, see the description of the pfCallerFreeProvOrNCryptKey parameter.
+        /// </param>
+        /// <param name="pdwKeySpec">The address of a DWORD variable that receives additional information about the key. This can be one of <see cref="KeySpec"/> values.</param>
+        /// <param name="pfCallerFreeProvOrNCryptKey">
+        /// The address of a BOOL variable that receives a value that indicates whether the caller must free the handle returned in the <paramref name="phCryptProvOrNCryptKey"/> variable.
+        /// This receives FALSE if any of the following is true:
+        /// - Public key acquisition or comparison fails.
+        /// - The <paramref name="dwFlags"/> parameter contains the <see cref="CryptAcquireCertificatePrivateKeyFlags.CRYPT_ACQUIRE_CACHE_FLAG"/> flag.
+        /// - The <paramref name="dwFlags"/> parameter contains the <see cref="CryptAcquireCertificatePrivateKeyFlags.CRYPT_ACQUIRE_USE_PROV_INFO_FLAG"/> flag, the certificate context property is set to <see cref="CERT_PROP_ID.CERT_KEY_PROV_INFO_PROP_ID"/> with the <see cref="CRYPT_KEY_PROV_INFO"/> structure, and the <paramref name="dwFlags"/> member of the <see cref="CRYPT_KEY_PROV_INFO"/> structure is set to CERT_SET_KEY_CONTEXT_PROP_ID.
+        /// If this variable receives FALSE, the calling application must not release the handle returned in the <paramref name="phCryptProvOrNCryptKey"/> variable.
+        /// The handle will be released on the last free action of the certificate context.
+        /// If this variable receives TRUE, the caller is responsible for releasing the handle returned in the <paramref name="phCryptProvOrNCryptKey"/> variable.
+        /// If the <paramref name="pdwKeySpec"/> variable receives the <see cref="KeySpec.CERT_NCRYPT_KEY_SPEC"/> flag, the handle must be released by passing it to the NCryptFreeObject function;
+        /// otherwise, the handle is released by passing it to the CryptReleaseContext function.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// If the function fails, the return value is zero.
+        /// </returns>
+        [DllImport(nameof(Crypt32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool CryptAcquireCertificatePrivateKey(
+                IntPtr pCert,
```

Comment:
Because that's how it's typed in the header file and documented on MSDN.
It doesn't seem to differ from your counter-example.  I think it's just that that method already existed and we never defined `CERT_CONTEXT` up to this point. So it's a gap that we _might_ want to fill before we make it any wider.